settings.h

```diff
@@ -25,8 +25,13 @@
 #include "system.h"
 
 
-// define GRBL version and build
-#define GRBL_VERSION "0.9K.3.7"
+// This is a bit of compiler magic to ensure that we don't
+// accidentally optimize away variables that are only used by external
+// tools (version numbers, ident blocks, etc)
+#define ALWAYS_KEEP(v) __asm__ __volatile__("" :: "m" (v))
+
+#define STR(s) #s
+#define XSTR(s) STR(s)
```

Comment:
Why define both `STR` and `XSTR`? They appear to do the same thing, and outside of the implementation of `XSTR`, `STR` appears unused.